src/nvim/eval.c

```diff
@@ -20477,6 +20477,27 @@ void set_selfdict(typval_T *rettv, dict_T *selfdict)
   }
 }
 
+// Turn a typeval into a string.  Similar to tv_get_string_buf() but uses
+// string() on Dict, List, etc.
+static const char *tv_stringify(typval_T *varp, char *buf)
+  FUNC_ATTR_NONNULL_ALL
+{
+  if (varp->v_type == VAR_LIST
+      || varp->v_type == VAR_DICT
+      || varp->v_type == VAR_FUNC
+      || varp->v_type == VAR_PARTIAL
+      || varp->v_type == VAR_FLOAT) {
+    typval_T tmp;
+
+    f_string(varp, &tmp, NULL);
+    const char *const res = tv_get_string_buf(&tmp, buf);
```

Comment:
Couldn't ignore the return value. Doesn't seem to make a diffference on the tests.

Reply:
reading the docstring for `tv_get_string_buf` makes me wonder. 

Does `(void)res` avoid the warning? 

```c
    const char *const res = tv_get_string_buf(&tmp, buf);
    (void)res;
    tv_clear(varp);
    *varp = tmp;
    return tmp.vval.v_string;
```

Reply:
That works on my machine.

Reply:
Why not `return encode_tv2string(&varp[0], NULL)` to skip the temporary variables? End result is the same.

Reply:
Nvrmnd. I can't figure out how to simplify it without the memory leak.